examples/levy-ssm/gamma_process.jl

```diff
@@ -0,0 +1,176 @@
+using Random
+using Plots
+using Distributions
+using AdvancedPS
+using LinearAlgebra
+
+struct GammaProcess
+    C::Float64
+    beta::Float64
+end
+
+struct GammaPath{T}
+    jumps::Vector{T}
+    times::Vector{T}
+end
+
+struct LangevinDynamics{T}
+    A::Matrix{T}
+    L::Vector{T}
+    θ::T
+    H::Vector{T}
+    σe::T
+end
+
+struct NormalMeanVariance{T}
+    μ::T
+    σ::T
+end
+
+function simulate(
+    process::GammaProcess,
+    rate::Float64,
+    start::Float64,
+    finish::Float64,
+    truncation::Float64,
+    t0::Float64=0.0,
+)
+    # Simulate the jumps
+    let beta = process.beta, C = process.C
+        jumps = Float64[]
+        last_jump = Inf
+        t = t0
+        truncated = last_jump < truncation
+        while !truncated
+            t += rand(Exponential(1.0 / rate))
+            xi = 1.0 / (beta * (exp(t) / C - 1))
+            prob = (1.0 + beta * xi) * exp(-beta * xi)
+            if rand() < prob
+                push!(jumps, xi)
+                last_jump = xi
+            end
+            truncated = last_jump < truncation
```

Comment:
What paper is this step coming from. To my understanding, in the original LevySSM paper we truncate the series when `t` (the Poisson epochs) exceed some truncation parameter `c` multiplied by `dt`.

Reply:
I think Yaman uses a more fancy adaptive truncation scheme